### DISCUSSION TITLE
How do I clean up expired files?

### DISCUSSION THREAD

**Discussion by ivancarlosti** [pglombardo/PasswordPusher#2600] [Q&A] [answered]
Hi masters, I am looking to cleanup files expired, and looks like it's made when a URL is acessed after it's expiration. (If I do not got it wrong, bad english, sorry) [issue](https://github.com/pglombardo/PasswordPusher/issues/468), [workaround](https://github.com/ivancarlosti/ivancarlos/wiki/Cleaning-up-PasswordPusher-expired-pushes)

I run PP in a docker, so, if my understanding is correctly, I just need to run periodically these commands:

```
rake daily_expiration
rake delete_expired_and_anonymous
```

Question: How to schedule it on PP in docker? :)

Thanks a lot and regards

**Comment by pglombardo** [comment]
Hi @ivancarlosti - the only way currently is to run them manually from time to time.

I do have #2601 though which introduces a new `pwpush-worker` container.  You launch that container with the same environment and settings as the main `pwpush` container and it will automatically run maintenance jobs (plus more).

It should be out/available in the next week or so.

**Comment by pglombardo** [comment, marked as answer]
As an update for any future visitors here:  Password Pusher now has a beta (and still to be documented) `pwpush-worker` container. 

Even if you don't run that, you can [manually run cleanup jobs through the Administration dashboard](https://docs.pwpush.com/docs/background-tasks/#running-the-background-tasks).
